base_vat_optional_vies/tests/test_res_partner.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Tecnativa - Antonio Espinosa
+# Copyright 2016 Tecnativa - Sergio Teruel
+# Copyright 2016 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestResPartner(TransactionCase):
+
+    def setUp(self):
+        super(TestResPartner, self).setUp()
+        self.m_partner = self.env['res.partner']
+        self.m_company = self.env['res.company']
+        self.company = self.m_company.browse(self.ref('base.main_company'))
+        self.partner = self.m_partner.browse(self.ref('base.res_partner_1'))
+
+    def test_split_vat(self):
+        cases = (
+            # vat, country, => vat_country, vat_number
+            ('ESB12345678', False, 'ES', 'B12345678'),
+            ('B12345678', False, 'XX', 'B12345678'),
+            ('1EB12345678', False, 'XX', '1EB12345678'),
+            ('ESB12345678', 'DE', 'ES', 'B12345678'),
+            ('B12345678', 'ES', 'ES', 'B12345678'),
+        )
+        for vat, country, vat_country, vat_number in cases:
+            res_country, res_number = self.m_partner._split_vat(vat, country)
+            self.assertEqual(res_country.upper(), vat_country)
+            self.assertEqual(res_number, vat_number)
+
+    def _test_validate_vat(self, cases):
+        for vat, res_vat, res_vies in cases:
+            self.partner.vat = vat
+            self.assertEqual(self.partner.vat.upper(), res_vat)
+            self.assertEqual(self.partner.vies_passed, res_vies)
+
+    # AEA: Can't use this test in Travis, VIES checking returns always False
```

Comment:
Remove this if not used, but try again to see. Maybe you can mock the call to VIES checking